src/runtime_src/core/tools/xbmgmt2/SubCmdConfigure.cpp

```diff
@@ -0,0 +1,515 @@
+      // For common options
+      for (auto& configOption : configOptions.options()) {
+        conflictingOptions(vm, loadConfigOption->long_name(), configOption->long_name());
+      }
+      // For hidden options
+      for (auto& configHiddenOption : configHiddenOptions.options()) {
+        conflictingOptions(vm, loadConfigOption->long_name(), configHiddenOption->long_name());
+      }
+    }
+
+    // Check the options
+    // -- process "help" option -----------------------------------------------
+    if (help) {
+        printHelp(commonOptions, hiddenOptions);
+        return;
+    }
+
+    // -- process "device" option -----------------------------------------------
+    if(devices.empty()) {
+        std::cerr << "ERROR: Please specify a single device using --device option" << "\n";
+        printHelp(commonOptions, hiddenOptions);
+        throw xrt_core::error(std::errc::operation_canceled);
+    }
+
+    // Collect all of the devices of interest
+    std::set<std::string> deviceNames;
+    xrt_core::device_collection deviceCollection;
+    for (const auto & deviceName : devices) 
+        deviceNames.insert(boost::algorithm::to_lower_copy(deviceName));
+
+    try {
+        XBU::collect_devices(deviceNames, false /*inUserDomain*/, deviceCollection);
+    } catch (const std::runtime_error& e) {
+        // Catch only the exceptions that we have generated earlier
+        std::cerr << boost::format("ERROR: %s\n") % e.what();
+        throw xrt_core::error(std::errc::operation_canceled);
+    }
+
+    // enforce 1 device specification
+    if(deviceCollection.size() != 1) {
+        std::cerr << "ERROR: Please specify a single device. Multiple devices are not supported" << "\n\n";
+        printHelp(commonOptions, hiddenOptions);
+        throw xrt_core::error(std::errc::operation_canceled);
+    }
+
+    std::shared_ptr<xrt_core::device>& workingDevice = deviceCollection[0];
+
+    // Load Config commands
+    // -- process "input" option -----------------------------------------------
+    if (!path.empty()) {
+        if (!boost::filesystem::exists(path)) {
+            std::cerr << boost::format("ERROR: Input file does not exist: '%s'") % path << "\n\n";
+            throw xrt_core::error(std::errc::operation_canceled);
+        }
+
+        if(boost::filesystem::extension(path).compare(".ini") != 0) {
+            std::cerr << boost::format("ERROR: Input file should be an INI file: '%s'") % path << "\n\n";
+            throw xrt_core::error(std::errc::operation_canceled);
+        }
+
+        try {
+            load_config(workingDevice, path);
+            std::cout << "Config has been successfully loaded" << std::endl;
+            return;
+        } catch (const std::runtime_error& e) {
+            // Catch only the exceptions that we have generated earlier
+            std::cout << boost::format("ERROR: %s\n") % e.what();
+            throw xrt_core::error(std::errc::operation_canceled);
+        }
```

Comment:
should we add a return here?

Reply:
We do not need to put returns after a throw. We return as part of the throw :)

Reply:
Oh I missed the return in the try block